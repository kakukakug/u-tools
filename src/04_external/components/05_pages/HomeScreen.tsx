import React from "react";
import {
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";
import { useNavigation } from "@react-navigation/native";

import { Colors } from "../../styles/Colors";

export const HomeScreen = () => {
  const navigation = useNavigation();

  const goIcons = () => {
    navigation.navigate("Icons");
  };
  return (
    <View style={styles.container}>
      <ScrollView
        style={styles.container}
        contentContainerStyle={styles.contentContainer}>
        <View style={styles.menuContainer}>
          <TouchableOpacity onPress={goIcons} style={styles.linkButton}>
            <Text style={styles.buttonText}>
              React Native Vector Icons の アイコンプレビュー
            </Text>
          </TouchableOpacity>
        </View>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.background,
  },
  contentContainer: {},
  menuContainer: {
    marginTop: 15,
    alignItems: "center",
  },
  linkButton: {
    margin: 5,
    padding: 15,
  },
  buttonText: {
    fontSize: 14,
    color: "#2e78b7",
  },
});
