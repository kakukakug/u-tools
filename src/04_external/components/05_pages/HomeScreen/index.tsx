import React from "react";
import { Text, ScrollView, StyleSheet, View } from "react-native";
import { Button } from "react-native-paper";
import { useNavigation } from "@react-navigation/native";

import { Colors } from "../../../styles/Colors";

export const HomeScreen = () => {
  const navigation = useNavigation();

  const goFlexPreview = () => {
    navigation.navigate("FlexPreview");
  };
  const goIcons = () => {
    navigation.navigate("Icons");
  };
  return (
    <View style={styles.container}>
      <View style={styles.title}>
        <Text style={styles.titleText}>React Native U-tools</Text>
      </View>
      <ScrollView
        style={styles.container}
        contentContainerStyle={styles.contentContainer}>
        <View style={styles.menuContainer}>
          <Button
            uppercase={false}
            onPress={goIcons}
            mode="outlined"
            style={styles.linkButton}>
            <Text style={styles.buttonText}>@expo/vector-icons Preview</Text>
          </Button>
          <Button
            uppercase={false}
            onPress={goFlexPreview}
            mode="outlined"
            style={styles.linkButton}>
            <Text style={styles.buttonText}>
              React Native FlexBox Style Creator
            </Text>
          </Button>
        </View>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.background,
  },
  contentContainer: {},
  menuContainer: {
    marginTop: 15,
    alignItems: "center",
  },
  linkButton: {
    margin: 10,
  },
  buttonText: {
    fontSize: 20,
    color: "#2e78b7",
  },
  title: { marginVertical: 10, alignSelf: "center" },
  titleText: { fontSize: 38, color: Colors.text },
});
