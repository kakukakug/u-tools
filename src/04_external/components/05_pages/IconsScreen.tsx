import React from "react";
import {
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  TextInput,
} from "react-native";
import { useNavigation } from "@react-navigation/native";

import Icon from "react-native-vector-icons/MaterialIcons";

import { Colors } from "../../styles/Colors";

export const IconsScreen = () => {
  const navigation = useNavigation();

  const keys = Object.keys(MaterialIconsGlyphs);
  console.log(keys);
  return (
    <View style={styles.container}>
      <View style={styles.title}>
        <Text style={styles.titleText}>アイコンプレビュー</Text>
      </View>
      <View style={styles.main}>
        <View style={styles.sidebar}>
          <View style={styles.selectIconSet}>
            <Text style={styles.iconSetText}>アイコンプレビュー</Text>
            <Text style={styles.iconSetText}>アイコンプレビュー</Text>
            <Text style={styles.iconSetText}>アイコンプレビュー</Text>
            <Text style={styles.iconSetText}>アイコンプレビュー</Text>
            <Text style={styles.iconSetText}>アイコンプレビュー</Text>
            <Text style={styles.iconSetText}>アイコンプレビュー</Text>
            <Text style={styles.iconSetText}>アイコンプレビュー</Text>
            <Text style={styles.iconSetText}>アイコンプレビュー</Text>
          </View>
          <View style={styles.code}>
            <Text style={styles.codeText}>CODE</Text>
            <Text style={styles.codeText}>CODE</Text>
            <Text style={styles.codeText}>CODE</Text>
            <Text style={styles.codeText}>CODE</Text>
          </View>
        </View>
        <ScrollView
          style={styles.iconView}
          contentContainerStyle={styles.contentContainer}>
          <View style={styles.search}>
            <TextInput />
          </View>
          <View style={styles.preview}>
            <TouchableOpacity onPress={() => {}} style={styles.iconButton}>
              <Icon size={13} name="info" />
              <Text style={styles.iconName}>info</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => {}} style={styles.iconButton}>
              <Icon size={13} name="info" />
              <Text style={styles.iconName}>info</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => {}} style={styles.iconButton}>
              <Icon size={13} name="info" />
              <Text style={styles.iconName}>info</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => {}} style={styles.iconButton}>
              <Icon size={13} name="info" />
              <Text style={styles.iconName}>info</Text>
            </TouchableOpacity>
          </View>
        </ScrollView>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.background,
    padding: 20,
  },
  title: { padding: 10 },
  titleText: { fontSize: 20 },
  main: {
    flex: 1,
    flexDirection: "row",
  },
  sidebar: {
    padding: 20,
    borderColor: "#888",
    borderWidth: 1,
    border: "solid",
  },
  selectIconSet: {
    padding: 10,
    borderColor: "#888",
    borderWidth: 1,
    border: "solid",
  },
  iconSetText: {},
  code: {
    backgroundColor: "#666",
    padding: 20,
  },
  codeText: {
    backgroundColor: "#aaa",
    Color: "#333",
    padding: 6,
  },
  iconView: {
    padding: 20,
    flex: 1,
  },
  contentContainer: {},
  search: {
    backgroundColor: "#aaa",
    padding: 6,
  },
  preview: {
    flex: 1,
    padding: 20,
  },
  iconButton: {
    margin: 4,
    padding: 10,
    borderColor: "#888",
    borderWidth: 1,
    border: "solid",
  },
  iconName: {},
});
