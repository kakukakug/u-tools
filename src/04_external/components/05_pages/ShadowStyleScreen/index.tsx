import React, { useState } from "react";
import { TextInput, ScrollView, StyleSheet, Text, View } from "react-native";
import Slider from "@react-native-community/slider";

import { Colors } from "../../../styles/Colors";

import { Console } from "../../01_atoms/Console";
import { PageTitle } from "../../01_atoms/PageTitle";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.background,
  },
  scrollContainer: {
    flex: 1,
  },
  contentContainer: {
    flex: 1,
    alignItems: "center",
  },
  main: {
    flexDirection: "row",
    justifyContent: "center",

    flexWrap: "wrap",
  },
  settingContainer: {
    margin: 10,
    padding: 10,
    backgroundColor: Colors.surface,
  },
  previewContainer: {
    margin: 10,
    paddingTop: 10,
    paddingHorizontal: 100,
    paddingBottom: 100,
    backgroundColor: Colors.white,
    borderColor: Colors.surface,
    borderWidth: 2,
  },
  subTitleText: {
    fontSize: 26,
    color: Colors.text,
    textAlign: "center",
    paddingTop: 6,
  },
  osText: {
    fontSize: 22,
    color: Colors.text,
    paddingVertical: 6,
    marginTop: 16,
  },
  previewBox: {
    backgroundColor: Colors.surface,
    padding: 40,
    alignSelf: "center",
  },
  textInput: {
    borderColor: Colors.surface,
    borderWidth: 2,
    backgroundColor: Colors.white,
    borderRadius: 4,
    padding: 4,
    marginTop: 4,
  },
  inputView: {
    backgroundColor: Colors.white,
    borderRadius: 4,
    padding: 10,
    margin: 4,
  },
  propertyName: {
    fontSize: 16,
    color: Colors.text,
    marginBottom: 4,
    marginTop: 2,
  },
  propertyValue: {
    fontSize: 16,
    color: Colors.icon,
    marginLeft: 10,
  },
});

export const ShadowStyleScreen = () => {
  const [shadowColor, setShadowColor] = useState("#333");
  const [shadowOffsetX, setShadowOffsetX] = useState(8);
  const [shadowOffsetY, setShadowOffsetY] = useState(8);
  const [shadowOpacity, setShadowOpacity] = useState(1);
  const [shadowRadius, setShadowRadius] = useState(0);

  const consoleText = `  componentShadow: {
    shadowColor: "${shadowColor}",
    shadowOffset: { width: ${shadowOffsetX}, height: ${shadowOffsetY} },
    shadowOpacity: ${shadowOpacity},
    shadowRadius: ${shadowRadius},
    elevation: 1,
  },`;

  return (
    <View style={styles.container}>
      <ScrollView
        style={styles.scrollContainer}
        contentContainerStyle={styles.contentContainer}>
        <PageTitle title="React Native Shadow Simulator" />
        <View style={styles.main}>
          <View style={styles.settingContainer}>
            <Text style={styles.subTitleText}>Property selecter</Text>
            <View>
              <Text style={styles.osText}>ios</Text>
              <View style={styles.inputView}>
                <Text>shadowColor</Text>
                <TextInput
                  style={styles.textInput}
                  placeholder="#333"
                  value={shadowColor}
                  onChangeText={setShadowColor}
                />
              </View>
              <View>
                <View style={styles.inputView}>
                  <Text style={styles.propertyName}>
                    shadowOffset-width
                    <Text style={styles.propertyValue}>
                      {`{ width: ${shadowOffsetX}, height: ${shadowOffsetY} }`}
                    </Text>
                  </Text>
                  <Slider
                    minimumValue={-50}
                    maximumValue={50}
                    minimumTrackTintColor={Colors.surface}
                    maximumTrackTintColor={Colors.text}
                    onValueChange={setShadowOffsetX}
                    value={shadowOffsetX}
                    step={1}
                  />
                  <Text style={styles.propertyName}>shadowOffset-height</Text>
                  <Slider
                    minimumValue={-50}
                    maximumValue={50}
                    minimumTrackTintColor={Colors.surface}
                    maximumTrackTintColor={Colors.text}
                    onValueChange={setShadowOffsetY}
                    value={shadowOffsetY}
                    step={1}
                  />
                </View>
              </View>
              <View>
                <View style={styles.inputView}>
                  <Text style={styles.propertyName}>
                    shadowOpacity
                    <Text style={styles.propertyValue}>{shadowOpacity}</Text>
                  </Text>
                  <Slider
                    minimumValue={0}
                    maximumValue={1}
                    minimumTrackTintColor={Colors.surface}
                    maximumTrackTintColor={Colors.text}
                    onValueChange={setShadowOpacity}
                    value={shadowOpacity}
                    step={0.01}
                  />
                </View>
              </View>
              <View>
                <View style={styles.inputView}>
                  <Text style={styles.propertyName}>
                    shadowRadius
                    <Text style={styles.propertyValue}>{shadowRadius}</Text>
                  </Text>
                  <Slider
                    minimumValue={0}
                    maximumValue={50}
                    minimumTrackTintColor={Colors.surface}
                    maximumTrackTintColor={Colors.text}
                    onValueChange={setShadowRadius}
                    value={shadowRadius}
                    step={1}
                  />
                </View>
              </View>
            </View>
            <View>
              <Text style={styles.osText}>android</Text>
              <View style={styles.inputView}>
                <Text>
                  {`android の場合は elevation で指定します。
Webでは再現できなかったので表示していません。`}
                </Text>
              </View>
            </View>
          </View>
          <View style={styles.previewContainer}>
            <Text style={styles.subTitleText}>preview</Text>
            <View>
              <Text style={styles.osText}>ios</Text>
              <View
                style={[
                  styles.previewBox,
                  {
                    shadowColor: shadowColor,
                    shadowOffset: {
                      width: shadowOffsetX,
                      height: shadowOffsetY,
                    },
                    shadowOpacity: shadowOpacity,
                    shadowRadius: shadowRadius,
                  },
                ]}>
                <Text>hello</Text>
              </View>
            </View>
          </View>
        </View>
      </ScrollView>
      <Console consoleText={consoleText} />
    </View>
  );
};
