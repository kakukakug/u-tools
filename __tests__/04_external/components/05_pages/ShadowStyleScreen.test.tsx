import React from "react";
import { render } from "@testing-library/react-native";

import { ShadowStyleScreen } from "../../../../src/04_external/components/05_pages/ShadowStyleScreen";

describe("ShadowStyleScreen", () => {
  it(`ShadowStyleScreenを描写`, () => {
    const component = render(<ShadowStyleScreen />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
