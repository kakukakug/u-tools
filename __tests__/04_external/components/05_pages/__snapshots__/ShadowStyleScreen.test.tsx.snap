// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShadowStyleScreen ShadowStyleScreenを描写 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#f9f9f9",
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "alignItems": "center",
            "flex": 1,
          }
        }
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "marginVertical": 10,
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#333",
                  "fontSize": 38,
                }
              }
            >
              React Native Shadow Simulator
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ccc",
                  "margin": 10,
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#333",
                    "fontSize": 26,
                    "paddingTop": 6,
                    "textAlign": "center",
                  }
                }
              >
                Property selecter
              </Text>
              <View>
                <Text
                  style={
                    Object {
                      "color": "#333",
                      "fontSize": 22,
                      "marginTop": 16,
                      "paddingVertical": 6,
                    }
                  }
                >
                  ios
                </Text>
                <View
                  style={
                    Object {
                      "backgroundColor": "#fffffe",
                      "borderRadius": 4,
                      "margin": 4,
                      "padding": 14,
                    }
                  }
                >
                  <Text>
                    shadowColor
                  </Text>
                  <TextInput
                    allowFontScaling={true}
                    onChangeText={[Function]}
                    placeholder="#333"
                    rejectResponderTermination={true}
                    style={
                      Object {
                        "backgroundColor": "#fffffe",
                        "borderColor": "#ccc",
                        "borderRadius": 4,
                        "borderWidth": 2,
                        "marginTop": 4,
                        "padding": 4,
                      }
                    }
                    underlineColorAndroid="transparent"
                    value="#333"
                  />
                </View>
                <View>
                  <View
                    style={
                      Object {
                        "backgroundColor": "#fffffe",
                        "borderRadius": 4,
                        "margin": 4,
                        "padding": 14,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#333",
                          "fontSize": 16,
                          "marginBottom": 4,
                          "marginTop": 2,
                        }
                      }
                    >
                      shadowOffset-width
                      <Text
                        style={
                          Object {
                            "color": "#666",
                            "flex": 1,
                            "fontSize": 16,
                            "marginLeft": 6,
                          }
                        }
                      >
                        { width: 8, height: 8 }
                      </Text>
                    </Text>
                    <View
                      aria-label="Slider"
                      colorScheme="blue"
                      dataSet={Object {}}
                      maxValue={50}
                      minValue={-50}
                      onChange={[Function]}
                      sliderSize={5}
                      step={1}
                      style={
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "width": "100%",
                        }
                      }
                      thumbSize={5}
                      value={
                        Array [
                          8,
                        ]
                      }
                    >
                      <View
                        accessible={true}
                        collapsable={false}
                        dataSet={Object {}}
                        focusable={true}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onHoverIn={[Function]}
                        onHoverOut={[Function]}
                        onLayout={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "height": 20,
                            "justifyContent": "center",
                            "paddingBottom": 12,
                            "paddingTop": 12,
                            "width": "100%",
                          }
                        }
                      >
                        <View
                          colorScheme="blue"
                          dataSet={Object {}}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#dbeafe",
                                "borderRadius": 8,
                                "height": 20,
                                "overflow": "hidden",
                                "width": 20,
                              },
                              Object {
                                "height": 5,
                                "width": "100%",
                              },
                            ]
                          }
                        >
                          <View
                            colorScheme="blue"
                            dataSet={Object {}}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "#2563eb",
                                  "height": 20,
                                  "left": 0,
                                  "position": "absolute",
                                  "width": 20,
                                },
                                Array [
                                  undefined,
                                  Object {
                                    "height": 5,
                                    "width": "57.99999999999999%",
                                  },
                                ],
                              ]
                            }
                          />
                        </View>
                      </View>
                      <View
                        colorScheme="blue"
                        dataSet={Object {}}
                        onMoveShouldSetResponder={[Function]}
                        onMoveShouldSetResponderCapture={[Function]}
                        onResponderEnd={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderReject={[Function]}
                        onResponderRelease={[Function]}
                        onResponderStart={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        onStartShouldSetResponderCapture={[Function]}
                        scaleOnPressed={1.2}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#2563eb",
                              "borderRadius": 99999,
                              "height": 20,
                              "justifyContent": "center",
                              "position": "absolute",
                              "width": 20,
                              "zIndex": 999,
                            },
                            Array [
                              Object {
                                "bottom": undefined,
                                "left": "57.99999999999999%",
                                "transform": Array [
                                  Object {
                                    "translateX": -10,
                                  },
                                  Object {
                                    "scale": 1,
                                  },
                                ],
                              },
                              undefined,
                            ],
                          ]
                        }
                      />
                    </View>
                    <Text
                      style={
                        Object {
                          "color": "#333",
                          "fontSize": 16,
                          "marginBottom": 4,
                          "marginTop": 2,
                        }
                      }
                    >
                      shadowOffset-height
                    </Text>
                    <View
                      aria-label="Slider"
                      colorScheme="blue"
                      dataSet={Object {}}
                      maxValue={50}
                      minValue={-50}
                      onChange={[Function]}
                      sliderSize={5}
                      step={1}
                      style={
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "width": "100%",
                        }
                      }
                      thumbSize={5}
                      value={
                        Array [
                          8,
                        ]
                      }
                    >
                      <View
                        accessible={true}
                        collapsable={false}
                        dataSet={Object {}}
                        focusable={true}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onHoverIn={[Function]}
                        onHoverOut={[Function]}
                        onLayout={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "height": 20,
                            "justifyContent": "center",
                            "paddingBottom": 12,
                            "paddingTop": 12,
                            "width": "100%",
                          }
                        }
                      >
                        <View
                          colorScheme="blue"
                          dataSet={Object {}}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#dbeafe",
                                "borderRadius": 8,
                                "height": 20,
                                "overflow": "hidden",
                                "width": 20,
                              },
                              Object {
                                "height": 5,
                                "width": "100%",
                              },
                            ]
                          }
                        >
                          <View
                            colorScheme="blue"
                            dataSet={Object {}}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "#2563eb",
                                  "height": 20,
                                  "left": 0,
                                  "position": "absolute",
                                  "width": 20,
                                },
                                Array [
                                  undefined,
                                  Object {
                                    "height": 5,
                                    "width": "57.99999999999999%",
                                  },
                                ],
                              ]
                            }
                          />
                        </View>
                      </View>
                      <View
                        colorScheme="blue"
                        dataSet={Object {}}
                        onMoveShouldSetResponder={[Function]}
                        onMoveShouldSetResponderCapture={[Function]}
                        onResponderEnd={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderReject={[Function]}
                        onResponderRelease={[Function]}
                        onResponderStart={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        onStartShouldSetResponderCapture={[Function]}
                        scaleOnPressed={1.2}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#2563eb",
                              "borderRadius": 99999,
                              "height": 20,
                              "justifyContent": "center",
                              "position": "absolute",
                              "width": 20,
                              "zIndex": 999,
                            },
                            Array [
                              Object {
                                "bottom": undefined,
                                "left": "57.99999999999999%",
                                "transform": Array [
                                  Object {
                                    "translateX": -10,
                                  },
                                  Object {
                                    "scale": 1,
                                  },
                                ],
                              },
                              undefined,
                            ],
                          ]
                        }
                      />
                    </View>
                  </View>
                </View>
                <View>
                  <View
                    style={
                      Object {
                        "backgroundColor": "#fffffe",
                        "borderRadius": 4,
                        "margin": 4,
                        "padding": 14,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#333",
                          "fontSize": 16,
                          "marginBottom": 4,
                          "marginTop": 2,
                        }
                      }
                    >
                      shadowOpacity
                      <Text
                        style={
                          Object {
                            "color": "#666",
                            "flex": 1,
                            "fontSize": 16,
                            "marginLeft": 6,
                          }
                        }
                      >
                        1
                      </Text>
                    </Text>
                    <View
                      aria-label="Slider"
                      colorScheme="blue"
                      dataSet={Object {}}
                      maxValue={1}
                      minValue={0}
                      onChange={[Function]}
                      sliderSize={5}
                      step={0.01}
                      style={
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "width": "100%",
                        }
                      }
                      thumbSize={5}
                      value={
                        Array [
                          1,
                        ]
                      }
                    >
                      <View
                        accessible={true}
                        collapsable={false}
                        dataSet={Object {}}
                        focusable={true}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onHoverIn={[Function]}
                        onHoverOut={[Function]}
                        onLayout={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "height": 20,
                            "justifyContent": "center",
                            "paddingBottom": 12,
                            "paddingTop": 12,
                            "width": "100%",
                          }
                        }
                      >
                        <View
                          colorScheme="blue"
                          dataSet={Object {}}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#dbeafe",
                                "borderRadius": 8,
                                "height": 20,
                                "overflow": "hidden",
                                "width": 20,
                              },
                              Object {
                                "height": 5,
                                "width": "100%",
                              },
                            ]
                          }
                        >
                          <View
                            colorScheme="blue"
                            dataSet={Object {}}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "#2563eb",
                                  "height": 20,
                                  "left": 0,
                                  "position": "absolute",
                                  "width": 20,
                                },
                                Array [
                                  undefined,
                                  Object {
                                    "height": 5,
                                    "width": "100%",
                                  },
                                ],
                              ]
                            }
                          />
                        </View>
                      </View>
                      <View
                        colorScheme="blue"
                        dataSet={Object {}}
                        onMoveShouldSetResponder={[Function]}
                        onMoveShouldSetResponderCapture={[Function]}
                        onResponderEnd={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderReject={[Function]}
                        onResponderRelease={[Function]}
                        onResponderStart={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        onStartShouldSetResponderCapture={[Function]}
                        scaleOnPressed={1.2}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#2563eb",
                              "borderRadius": 99999,
                              "height": 20,
                              "justifyContent": "center",
                              "position": "absolute",
                              "width": 20,
                              "zIndex": 999,
                            },
                            Array [
                              Object {
                                "bottom": undefined,
                                "left": "100%",
                                "transform": Array [
                                  Object {
                                    "translateX": -10,
                                  },
                                  Object {
                                    "scale": 1,
                                  },
                                ],
                              },
                              undefined,
                            ],
                          ]
                        }
                      />
                    </View>
                  </View>
                </View>
                <View>
                  <View
                    style={
                      Object {
                        "backgroundColor": "#fffffe",
                        "borderRadius": 4,
                        "margin": 4,
                        "padding": 14,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#333",
                          "fontSize": 16,
                          "marginBottom": 4,
                          "marginTop": 2,
                        }
                      }
                    >
                      shadowRadius
                      <Text
                        style={
                          Object {
                            "color": "#666",
                            "flex": 1,
                            "fontSize": 16,
                            "marginLeft": 6,
                          }
                        }
                      >
                        0
                      </Text>
                    </Text>
                    <View
                      aria-label="Slider"
                      colorScheme="blue"
                      dataSet={Object {}}
                      maxValue={50}
                      minValue={0}
                      onChange={[Function]}
                      sliderSize={5}
                      step={1}
                      style={
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "width": "100%",
                        }
                      }
                      thumbSize={5}
                      value={
                        Array [
                          0,
                        ]
                      }
                    >
                      <View
                        accessible={true}
                        collapsable={false}
                        dataSet={Object {}}
                        focusable={true}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onHoverIn={[Function]}
                        onHoverOut={[Function]}
                        onLayout={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "height": 20,
                            "justifyContent": "center",
                            "paddingBottom": 12,
                            "paddingTop": 12,
                            "width": "100%",
                          }
                        }
                      >
                        <View
                          colorScheme="blue"
                          dataSet={Object {}}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#dbeafe",
                                "borderRadius": 8,
                                "height": 20,
                                "overflow": "hidden",
                                "width": 20,
                              },
                              Object {
                                "height": 5,
                                "width": "100%",
                              },
                            ]
                          }
                        >
                          <View
                            colorScheme="blue"
                            dataSet={Object {}}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "#2563eb",
                                  "height": 20,
                                  "left": 0,
                                  "position": "absolute",
                                  "width": 20,
                                },
                                Array [
                                  undefined,
                                  Object {
                                    "height": 5,
                                    "width": "0%",
                                  },
                                ],
                              ]
                            }
                          />
                        </View>
                      </View>
                      <View
                        colorScheme="blue"
                        dataSet={Object {}}
                        onMoveShouldSetResponder={[Function]}
                        onMoveShouldSetResponderCapture={[Function]}
                        onResponderEnd={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderReject={[Function]}
                        onResponderRelease={[Function]}
                        onResponderStart={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        onStartShouldSetResponderCapture={[Function]}
                        scaleOnPressed={1.2}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#2563eb",
                              "borderRadius": 99999,
                              "height": 20,
                              "justifyContent": "center",
                              "position": "absolute",
                              "width": 20,
                              "zIndex": 999,
                            },
                            Array [
                              Object {
                                "bottom": undefined,
                                "left": "0%",
                                "transform": Array [
                                  Object {
                                    "translateX": -10,
                                  },
                                  Object {
                                    "scale": 1,
                                  },
                                ],
                              },
                              undefined,
                            ],
                          ]
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
              <View>
                <Text
                  style={
                    Object {
                      "color": "#333",
                      "fontSize": 22,
                      "marginTop": 16,
                      "paddingVertical": 6,
                    }
                  }
                >
                  android
                </Text>
                <View
                  style={
                    Object {
                      "backgroundColor": "#fffffe",
                      "borderRadius": 4,
                      "margin": 4,
                      "padding": 14,
                    }
                  }
                >
                  <Text>
                    android の場合は elevation で指定します。
Webでは再現できなかったので表示していません。
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#fffffe",
                  "borderColor": "#ccc",
                  "borderWidth": 2,
                  "margin": 10,
                  "paddingBottom": 100,
                  "paddingHorizontal": 100,
                  "paddingTop": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#333",
                    "fontSize": 26,
                    "paddingTop": 6,
                    "textAlign": "center",
                  }
                }
              >
                preview
              </Text>
              <View>
                <Text
                  style={
                    Object {
                      "color": "#333",
                      "fontSize": 22,
                      "marginTop": 16,
                      "paddingVertical": 6,
                    }
                  }
                >
                  ios
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "backgroundColor": "#ccc",
                        "padding": 40,
                      },
                      Object {
                        "shadowColor": "#333",
                        "shadowOffset": Object {
                          "height": 8,
                          "width": 8,
                        },
                        "shadowOpacity": 1,
                        "shadowRadius": 0,
                      },
                    ]
                  }
                >
                  <Text>
                    hello
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        style={
          Object {
            "backgroundColor": "#333",
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={false}
          multiline={true}
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#666",
              "borderRadius": 4,
              "color": "#f9f9f9",
              "fontSize": 16,
              "height": 250,
              "marginHorizontal": 10,
              "marginTop": 10,
              "padding": 5,
              "textAlignVertical": "top",
            }
          }
          testID="console-text"
          underlineColorAndroid="transparent"
          value="  componentShadow: {
    shadowColor: \\"#333\\",
    shadowOffset: { width: 8, height: 8 },
    shadowOpacity: 1,
    shadowRadius: 0,
    elevation: 1,
  },"
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "opacity": 1,
              "padding": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fffffe",
                "fontSize": 18,
                "paddingVertical": 5,
                "textAlign": "center",
              }
            }
          >
            close
          </Text>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</RNCSafeAreaProvider>
`;
